### PR TITLE
add: integration test for NodeAffinity plugin in requeueing scenarios

### LINKED ISSUE #122305: Implement the integration tests for requeueing scenarios
/sig scheduling
/priority important-soon
/kind feature

Part of: https://github.com/kubernetes/kubernetes/issues/122284#issuecomment-1853346333.

We have few tests to check the scheduler's requeueing scenario in the integration test. 
We should have the one so that we can catch a bug like https://github.com/kubernetes/kubernetes/issues/122284, which only happens the scheduling queue is actually working.

---

Here's the list of plugins with QueueingHint.
Basically, I believe we can just pile up the test cases in the existing [TestCoreResourceEnqueue](https://github.com/kubernetes/kubernetes/blob/746f08a8da0a45e8b1a46362822db05cb11ed294/test/integration/scheduler/queue_test.go#L188).
Also, note that we have to cover major scenarios only, don't have to cover minor ones within the integration test.

**Please comment in this issue before starting to work on so that we can avoid many people from working on the same thing!**

- [x] InterPodAffinity @nayihz https://github.com/kubernetes/kubernetes/pull/127090
- [x] Dynamic resource allocation https://github.com/kubernetes/kubernetes/issues/122305#issuecomment-2336618368
- [x] NodeAffinity @tozastation https://github.com/kubernetes/kubernetes/pull/127139
- [x] NodePorts @saku3 https://github.com/kubernetes/kubernetes/pull/127130
- [x] NodeResourceFit @unvavo https://github.com/kubernetes/kubernetes/pull/127354
- [x] NodeUnschedulable @toshikish  https://github.com/kubernetes/kubernetes/pull/127406
- [ ] CSILimits @utam0k https://github.com/kubernetes/kubernetes/pull/127224
- [x] TopologySpread @saku3 https://github.com/kubernetes/kubernetes/pull/127399
- [x] TaintToleration @unvavo  https://github.com/kubernetes/kubernetes/pull/127923
- [ ] VolumeBinding @saku3  https://github.com/kubernetes/kubernetes/pull/127946
- [x] VolumeRestriction @utam0k https://github.com/kubernetes/kubernetes/pull/127892
- [x] VolumeZone @Gekko0114 https://github.com/kubernetes/kubernetes/pull/127353

### PR DESCRIPTION
**Pull request by tozastation**
#### What type of PR is this?

/kind feature

#### What this PR does / why we need it:

add integration test for NodeAffinity in requeueing scenarios

#### Which issue(s) this PR fixes:

Fixes #122305

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Review by sanposhiho — COMMENTED** [review]
/ok-to-test

File: test/integration/scheduler/queue_test.go [hunk 1]
```diff
@@ -254,6 +254,40 @@ func TestCoreResourceEnqueue(t *testing.T) {
 			wantRequeuedPods:          sets.New("pod2"),
 			enableSchedulingQueueHint: []bool{false, true},
 		},
+		{
```
Comment (sanposhiho) [inline test/integration/scheduler/queue_test.go:257]: Can you add one more Pod which have nodeaffinity, but shouldn't be requeued in this test case? Then you can remove the second test case.

You can add pod2 that has NodeAffinity but not match even after node's update. And then make sure pod1 is requeued, while pod2 isn't.

Comment (tozastation) [reply test/integration/scheduler/queue_test.go:257]: thanks for reviewing.
I changed the test code to combine them into one.

**Review by sanposhiho — APPROVED** [review]
/approve

**Pull request merged**